contextManager.tsx

```diff
@@ -0,0 +1,82 @@
+import { createContext, useState } from 'react';
+import { Card } from './mock_data';
+
+//Interface Type for my 
+export interface user {
+    id: string
+    firstName: string
+    lastName: string
+    email: string
+    phone: number
+    address: string
+    password: string
+    isMember: boolean
+    cards: Card[]
+    setName: (firstName: string, lastName: string) => void
+    setEmail: (email: string) => void
+    setPhone: (phone: number) => void
+    setAddress: (address: string) => void
+    setPassword: (password: string) => void
+    setIsMember: (isMember: boolean) => void
+    addCard: (newCard: Card) => void
+}
+
+const UserContext = createContext<user | null>(null);
+
+function UserProvider({ children }: any) {
```

Comment:
Separate the context provider and the user methods like setName, setEmail etc. I would create a hook called useUser(). Context manager stuff can go in this file and user stuff can go in the hook. You can refer to the Secure Assist user context or onboarding context.